componentes/addPostDb.tsx

```diff
@@ -0,0 +1,81 @@
+import { collection, addDoc } from "firebase/firestore/lite";
+import { db } from "../service/firebase";
+import {
+  Input,
+  Heading,
+  FormControl,
+  FormLabel,
+  Button,
+  Flex,
+} from "@chakra-ui/react";
+import { useState } from "react";
+
+export default function AddPost() {
+  const [namePost, setNamePost] = useState<string>("");
+  const [newsPost, setNewsPost] = useState<string>("");
+  const [imageUrl, setImageUrl] = useState<string>("");
+
+  async function addPostDb() {
+    try {
+      const docRef = await addDoc(collection(db, "post"), {
+        id: Math.random(),
+        namePost,
+        newsPost,
+        imageUrl,
```

Comment:
Uma dica pra facilitar a leitura é evitar criar esses objectos in-place e criar uma variavel pra ele `const newPost = { ... }`

Alem disso, por ser typescript, seria interessante ter um tipo para modelar esse dado. Ai ficaria algo tipo `const newPost: Post = { ... }`